src/control/cmd/daos/container.go

```diff
@@ -397,76 +394,50 @@ func (cmd *containerCreateCmd) contCreate() (string, error) {
 	if err != nil {
 		return "", err
 	}
+	contID := cmd.contUUID.String()
+	cContID := C.CString(contID)
+	defer freeString(cContID)
 
-	var contID string
-	if cmd.contUUID == uuid.Nil {
-		contID = cmd.contLabel
-	} else {
-		contID = cmd.contUUID.String()
-	}
-
-	cmd.Infof("Successfully created container %s", contID)
-	return contID, nil
-}
-
-func (cmd *containerCreateCmd) contCreateUNS() (string, error) {
-	var dattr C.struct_duns_attr_t
-
-	props, cleanupProps, err := cmd.getCreateProps()
-	if err != nil {
-		return "", err
+	cleanupContainer := func() {
+		rc := C.daos_cont_destroy(cmd.cPoolHandle, cContID, goBool2int(true), nil)
+		if err := daosError(rc); err != nil {
+			cmd.Noticef("Failed to clean-up container %v", err)
+		}
 	}
-	defer cleanupProps()
-	dattr.da_props = props
 
-	if !cmd.Type.Set {
-		return "", errors.New("container type is required for UNS")
-	}
-	dattr.da_type = cmd.Type.Type
+	if len(cmd.Attrs.ParsedProps) != 0 {
+		attrs := make(attrList, 0, len(cmd.Attrs.ParsedProps))
+		for key, val := range cmd.Attrs.ParsedProps {
+			attrs = append(attrs, &attribute{
+				Name:  key,
+				Value: []byte(val),
+			})
+		}
 
-	if cmd.poolUUID != uuid.Nil {
-		poolUUIDStr := C.CString(cmd.poolUUID.String())
-		defer freeString(poolUUIDStr)
-		C.uuid_parse(poolUUIDStr, &dattr.da_puuid[0])
-	}
-	if cmd.contUUID != uuid.Nil {
-		contUUIDStr := C.CString(cmd.contUUID.String())
-		defer freeString(contUUIDStr)
-		C.uuid_parse(contUUIDStr, &dattr.da_cuuid[0])
-	}
+		if err := cmd.openContainer(C.DAOS_COO_RW); err != nil {
+			cleanupContainer()
+			return "", errors.Wrapf(err, "failed to open new container %s", contID)
+		}
+		defer cmd.closeContainer()
 
-	if cmd.ChunkSize.Set {
```

Comment:
These CLI flags seem to be obsolete now, unless I'm missing something. For compatibility, they should be converted into `cmd.Attrs` entries. Even better would be to emit a deprecation warning and create a TODO ticket to remove them in the next+1 release.

Reply:
They are no obsolete CLI flags as they are still used in the merge of the old two functions `ContCreate()` and `ContCreateUNS()` in the new function `ContCreate()`.